src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -871,11 +879,21 @@ public override void VisitTernaryOperationSyntax(TernaryOperationSyntax syntax)
                 }
 
                 var expectedConditionType = LanguageConstants.Bool;
+                // if the condition is nullable, emit a fixable warning and proceed as if it had been non-nullable
+                if (TypeHelper.TryRemoveNullability(conditionType) is {} nonNullableConditionType && TypeValidator.AreTypesAssignable(nonNullableConditionType, expectedConditionType))
```

Comment:
The if condition repeats several times, maybe it's worth extracting a helper method?

Reply:
I like that idea; it should improve readability.